sklearn/svm/_base.py

```diff
@@ -217,7 +217,25 @@ def fit(self, X, y, sample_weight=None):
                 "X and y have incompatible shapes.\n"
                 + "X has %s samples, but y has %s." % (n_samples, y.shape[0])
             )
-
+        if self.kernel == "linear":
+            if self.gamma != "scale":
```

Comment:
@adrinjalali I think it's better to require a value of `gamma` to be clearly specified. Since the default value is already `None`, we might not want to have another default value when `None` is chosen. For me it's strange to have two default values for a single function call.